### DISCUSSION TITLE
[Projects Beta] Mass-assign issues to projects

### DISCUSSION THREAD

**Discussion by jwindmuller** [community/community#6765] [Projects and Issues] [answered]
When looking at the issues list in a repo it would be nice to be able to select them using the checkmarks and put them all into a project.

**Comment by RonnyB71** [comment]
Totally agree. I was looking to start with 104 issues in a repo, a major pain to add manually. I used regular issue search syntax in GitHub to identify the issues and would like to see the same syntax available when adding issues to a project - or just mass assign from the search result.

**Reply by atorres-eqrx** [reply]
What syntax did you use to identify issues that aren't in a project? 
From this [issue](https://github.com/isaacs/github/issues/1291) I was hoping to use `no:project` but that doesn't work for me.

**Reply by RonnyB71** [reply]
I couldn't - it was an initial job, then we needed to remember to manually add each and every one to et project as we create them. I was thinking about creating a small script that fetches issues based on a standard search query then compares them with issues in the project (I think you can use the GraphQL api for that as the regular REST api's doesn't seem to support Projects (Beta). You could use a GitHub Action for that as well.

**Comment by timoyates** [comment]
Thanks for the feedback, folks! We're exploring ideas along this vein at the moment. Stay tuned and cheers!

**Reply by Rick-Methot-NOAA** [reply]
It also would be helpful to add everything that is assigned to a specified Milestone.

**Comment by felipezarco** [comment]
I think at least two or three people to be assigned to the same issue would be a revolution in this already nice feature

![image](https://user-images.githubusercontent.com/11004919/139444827-8bd1b726-4b09-4118-bb23-44c20dc743e3.png)

**Reply by timoyates** [reply]
Hi @felipezarco are you referring to the Issues feature or even how Issues appear in projects? In either case, assigning multiple people should be supported already 😊

**Reply by ElijahLynn** [reply]
I've always wanted a "Person responsible" and another field to distinguish "People working on". Too often I don't know who is actually responsible for working on a ticket, including myself. Or maybe there is a way to have multiple people on the same field but be able to designate one as an owner or responsible.

**Comment by tdd** [comment]
In general, selecting issues to do some mass edition (e.g. set a field's value) would be a wonderful improvement. Right now having to manually set the same field over and over agian on all newly-added issues is infuriating.

**Comment by kristinarobinson** [comment]
I wanted to add a slight variation to this.  Would it be possible to auto-assign projects to issues based on a set of pre-defined criteria?  For example, when a set of labels "foo, bar, and baz" are all assigned to an issue contained in repo "general", then assigned to project "Project One".

**Comment by victorlin** [comment]
I've hacked together a python script to do this using the GitHub GraphQL API: https://gist.github.com/victorlin/8ce4fe5f7e9ab6c2aa6210f373a7857d

But, as @tdd mentioned, it's a pain to bulk-edit items once they've been added. This again could be scripted but that defeats the purpose of using the UI. I think I'll be holding off on using the Projects Beta until the bulk-edit features have been added.

**Comment by clapfootmark** [comment]
This is critical for my use case. Without the ability to bulk add issues, the new projects feature is largely useless. Are we really expected to hand add 100s of issues potentially?

**Comment by grgcnnr** [comment]
I'll add my voice to the chorus on this one. Setting up a beta project for a mature repo means adding a great many issues one by one, doing it from the project also means you have to sift through closed issues to find the one you're looking for

**Comment by MaximeBernard** [comment]
I didn't see this suggestion yet so let me suggest the ability to "import all issues from selected repositories" and a checkbox to "keep it sync".

**Reply by jleni** [reply]
I think this is the most important missing feature at this moment

**Reply by grgcnnr** [reply]
It's a bit cumbersome but you can do that currently with a GH action

**Reply by rappo** [reply]
@grgcnnr would you mind pointing me in the right direction? Which action did you use and how did you configure it to add/sync issues?

**Reply by grgcnnr** [reply]
Note. I haven't actually tried this but it;s in the official docs :)
https://docs.github.com/en/issues/trying-out-the-new-projects-experience/automating-projects

**Comment by VMois** [comment]
A similar feature already exists in classic Project Boards called "Add Cards". You open the sidebar, filter issues globally, and add them to the project.

**Reply by psifertex** [reply]
Yup, exactly. Just need that on the new UI.

**Comment by jleni** [comment]
When looking at the roadmap.. it seems that it will not be done in 2022.
Can someone confirm that?

**Reply by ElijahLynn** [reply]
Can you link the roadmap?

**Comment by omerbensaadon** [comment, marked as answer]
Hey, this has been released!

See change-log: https://github.blog/changelog/2022-04-07-the-new-github-issues-april-7th-update/

**Reply by m00sey** [reply]
If you have a filter selected (including no:project as shown in the release notes) it just offers a further filter, not add.

Add is only available if I have no additional filters applied (like filtering by label) so, largely still useless to me.

**Reply by archon810** [reply]
The Github Issues UI and the Projects Beta UI to add issues both max out at 25 issues - is there no way around this? Imagine a repo with hundreds of issues.

In the Issues UI, once I select the max amount of issues per page (25), there's no way to select all matching issues (like you can do in, say, Gmail) - you have to paginate page by page to add issues. Furthermore, even if some or all selected issues are already in the project I'm adding to, I still get "N issues have successfully been added to 1 project" even if all N of them are already in the project.

In the Projects Beta UI, there's no way to return more than 25 results. If I do "is:open is:issue", then add those 25 results, pressing Enter again makes the list empty rather than load the next 25. I have to close the Add UI, reopen it, do the search again, then 25 more show up, and rinse and repeat. This is even less user-friendly behavior than the Issue UI one above.

I hope the team can work on improving these aspects of the new Projects Beta.
